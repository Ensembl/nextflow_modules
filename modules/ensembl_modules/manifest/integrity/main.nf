process MANIFEST_INTEGRITY {
    tag "$meta.id"
    label 'process_low'

    // TODO nf-core: List required Conda package(s).
    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
    // conda "${moduleDir}/environment.yml"
    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     'https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE':
    //     'biocontainers/YOUR-TOOL-HERE' }"

    input:
        tuple val(meta), path(manifest_files)
    
    output:
        tuple val(meta), path("*.*", includeInputs: true), emit: all_files, optional: true
        tuple val(meta), path(integrity_file), emit: error_log, optional: true
        // path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
        def args = task.ext.args ?: ''
        def prefix = task.ext.prefix ?: "${meta.id}"
        integrity_file = "integrity.out"
        """
        manifest_check_integrity \
            --manifest_file ./manifest.json \
            --no_fail \
            $brc_mode \
            > $integrity_file
        
        # Only keep integrity file if there are errors to report
        if [ ! -s $integrity_file ]
            then rm $integrity_file
        fi
        """

    stub:
        def args = task.ext.args ?: ''
        def prefix = task.ext.prefix ?: "${meta.id}"
        integrity_file = "integrity.out"
        """
        echo "No change, don't create manifest error log"
        """
}
