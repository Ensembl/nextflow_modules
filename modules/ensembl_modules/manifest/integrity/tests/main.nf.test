nextflow_process {

    name "Test Process MANIFEST_INTEGRITY"
    script "modules/ensembl_modules/manifest/integrity/main.nf"
    process "MANIFEST_INTEGRITY"
    tag "modules"
    tag "modules_nfcore"
    tag "manifest/integrity"
    tag "manifest"

    test("Integrity check stub") {

        when {
            options "-stub"

            process {
                """
                // Just one GFF file
                
                input[0] = Channel.of(
                    '##gff-version 3',
                    'chr1\tID=gene1'
                    )
                    .collectFile(name: 'genes.gff3', newLine: true)
                    .map { file -> [ [ id:'test' ], file ] }
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
