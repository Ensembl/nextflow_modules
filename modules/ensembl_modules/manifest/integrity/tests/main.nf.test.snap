{
    "Integrity check stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genes.gff3:md5,5e2fd50352e7d2ebec2bdccaa352e541",
                            "versions.yml:md5,a0095dcbbbfdcbd9c2b17b3d36571d42"
                        ]
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    "versions.yml:md5,a0095dcbbbfdcbd9c2b17b3d36571d42"
                ],
                "all_files": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "genes.gff3:md5,5e2fd50352e7d2ebec2bdccaa352e541",
                            "versions.yml:md5,a0095dcbbbfdcbd9c2b17b3d36571d42"
                        ]
                    ]
                ],
                "error_log": [
                    
                ],
                "versions": [
                    "versions.yml:md5,a0095dcbbbfdcbd9c2b17b3d36571d42"
                ]
            }
        ],
        "timestamp": "2024-08-07T14:32:11+0000"
    }
}